source/extensions/filters/http/cache/cacheability_utils.cc

```diff
@@ -69,8 +69,14 @@ bool CacheabilityUtils::isCacheableResponse(const Http::ResponseHeaderMap& heade
       (headers.Date() && response_cache_control.max_age_.has_value()) ||
       headers.get(Http::Headers::get().Expires);
 
+  // TODO(cbdm): only allowing responses to be cacheable if there's no vary header or it is empty;
+  // need to implement the logic for using an allowlist.
+  const bool no_vary = (!headers.get(Http::Headers::get().Vary)) ||
```

Comment:
make a temp for `headers.get(Http::Headers::get().Vary` rather than calling that sequence twice.